### ISSUE TITLE
Missing hand of cards and messages after waiting to join

### ISSUE THREAD

**Issue by domcleal** [grifferz/pah-irc#143]
I attempted to join a game that was waiting for the Tsar to pick the winning card, was queued, then later the Tsar was forcibly resigned and I was dealt into the game.

<pre>
095833 <@Dominic> AgainstHumanity: dealmein
095833 <+AgainstHumanity> Dominic: Sorry, this round is complete and we're waiting on lanky to pick the winner. Stay on IRC and you'll be dealt in to the next round automatically.
105756 <@kfullert> AgainstHumanity: status
105756 <+AgainstHumanity> Waiting on lanky to pick the winning play. Round started about 1d55m ago. Idle punishment in about 1d9h.
105756 <+AgainstHumanity> The Card Tsar is lanky; current Black Card:
105757 <+AgainstHumanity> → How am I maintaining my relationship status?
... next day ...
195832 <+AgainstHumanity> I'm forcibly resigning lanky from the game due to idleness. Idle since 2015-04-16T16:27:34.
195832 <+AgainstHumanity> The new Card Tsar is kfullert. Time for the next Black Card:
195832 <+AgainstHumanity> → After the earthquake, Sean Penn brought _____ to the people of Haiti.
195832 <+AgainstHumanity> Dominic: You're in now, too!
215942 <+AgainstHumanity> 1 person recently played! We're currently waiting on plays from 3 people: kwyjibo, Dominic and Eckrall.
</pre>

When this happened, I didn't receive any private messages from the bot, and when I go to check my hand, it's entirely empty:

<pre>
222246 &lt;Dominic> #surrey-ot hand
222246 &lt;AgainstHumanity> Your White Cards in #surrey-ot:
</pre>

**Comment by grifferz** [comment]
The code path where the game is progressed by Card Tsar resignation has `topup_hands()` before `pick_new_tsar()` whereas every other path for ending a round has them in the other order.

People waiting to join are added in `pick_new_tsar()`, so they don't get their hand topped up through that path.
